addons/base_vat/models/res_partner.py

```diff
@@ -165,7 +165,7 @@ def check_vat(self):
             country = partner.commercial_partner_id.country_id
             if self._run_vat_test(partner.vat, country, partner.is_company) is False:
```

Comment:
Why not only doing this?
```suggestion
            if country.code and self._run_vat_test(partner.vat, country, partner.is_company) is False:
```
I cannot think of a country without a code that would benefit from the `_run_vat_test`.

Even the `xi` code from special regions used in specific context for Northern Ireland belongs to Great Britain (country code: `gb`)

Reply:
We cant do this, because of few test cases failed.

Reply:
@alsh-odoo 
Indeed. I'm in a discussion with the PO about the PR in Master, I'll see with him what's best in stable between "adding condition" vs "prevent user to delete the country code".